--- ../src-base/minecraft/net/minecraft/entity/passive/EntitySkeletonHorse.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntitySkeletonHorse.java
@@ -23,6 +23,9 @@
     private boolean field_184793_bU;
     private int field_184794_bV;
 
+    //CM
+    public String cm_name() { return "Skeleton Horse"; }
+
     public EntitySkeletonHorse(World p_i47295_1_)
     {
         super(p_i47295_1_);
